scripts/common/addon.sh

```diff
@@ -307,3 +309,49 @@ function addon_source() {
     # shellcheck disable=SC1090
     . "$DIR/addons/$name/$version/install.sh"
 }
+
+# addon_update_used_images takes an addon name and version as parameters.
+# optionally, a third 'pathoverride' parameter can be passed to specify a path to the addon's images file.
+# it checks the kurl-current-images configmap for a match, and adds those images to kurl-old-images.
+# it then adds the new addon version's images to the kurl-current-images map and removes previous addon version's images.
+function addon_update_used_images() {
+    local name=$1
+    local version=$2
+    local pathoverride=$3
+
+    if ! kubectl get configmap -n kurl kurl-current-images; then
+        kubectl create configmap -n kurl kurl-current-images
+    fi
+    if ! kubectl get configmap -n kurl kurl-old-images; then
+        kubectl create configmap -n kurl kurl-old-images
+    fi
+
+    # update kurl-old-images configmap with images from the previous addon version
```

Comment:
What if you were to store a list of images with the key `[addon]-[version]` and another key `[addon]` with value `[version]` to look up the previous version? Wouldn't that greatly simplify the logic in this function? This is only a suggestion.

Reply:
It might simplify the logic here, but I'm not sure it would simplify the logic overall.

That being said, I'm not sure if storing the list of images is even _useful_, so...

Reply:
(Because there's not really a way to delete a host containerd image from within a k8s pod)

I think we might just want to rely on the k8s automatic image eviction here, but there's always the risk it'll prune an image we're migrating to in airgap. Another reason to use a registry)

Reply:
if we'd rather solve images holistically im fine with that